File: clients/mediafx-py/python/tests/color.py
#!/usr/bin/env python
# Copyright (C) 2025 Andrew Wason
# SPDX-License-Identifier: GPL-3.0-or-later

import struct

from mediafx import MediaFX


def main() -> None:
    client = MediaFX()
    width, height = client.frame_size
    rendered_frame = bytearray(client.frame_bytecount)

    while True:
        (time, *_) = client.render_begin()
        r = int(time % 255)
        b = int((time + 128) % 255)
        pixel = struct.pack("=4B", r, 0, b, 255)
        for outpixel in range(0, len(rendered_frame), 4):
            rendered_frame[outpixel : outpixel + 4] = pixel
        client.render_finish(rendered_frame)


if __name__ == "__main__":
    main()

File: clients/mediafx_node/examples/color.ts
#!/usr/bin/env node
// Copyright (C) 2025 Andrew Wason
// SPDX-License-Identifier: GPL-3.0-or-later

const { MediaFX } = require("../index.js");

const client = new MediaFX();
// const [width, height] = client.frameSize;
const renderedFrame = new Uint8Array(client.frameBytecount);

while (true) {
  let [, r, g, b] = client.renderFrame();
  r = Math.floor(r * 255);
  g = Math.floor(g * 255);
  b = Math.floor(b * 255);
  for (let offset = 0; offset < renderedFrame.length; offset += 4) {
    renderedFrame[offset] = r;
    renderedFrame[offset + 1] = g;
    renderedFrame[offset + 2] = b;
    renderedFrame[offset + 3] = 255;
  }
  client.renderCommit(renderedFrame);
}
